test/cppsuite/test_harness/workload_validation.h

```diff
@@ -0,0 +1,395 @@
+     * that contains all the operations about the creation or deletion of collections during the
+     * test.
+     */
+    bool
+    validate(const std::string &operation_table_name, const std::string &schema_table_name)
+    {
+        WT_SESSION *session;
+        std::string collection_name;
+        /*
+         * Representation in memory of the collections at the end of the test. The first level is a
+         * map that contains collection names as keys. The second level is another map that contains
+         * the different key/value pairs within a given collection. If a collection yields to a null
+         * map of key/value pairs, this means the collection should not be present on disk. If a
+         * value from a key/value pair is null, this means the key should not be present in the
+         * collection on disk.
+         */
+        std::map<std::string, std::map<int, std::string *> *> collections;
+        /* Existing collections after the test. */
+        std::vector<std::string> created_collections;
+        bool is_valid;
+
+        session = connection_manager::instance().create_session();
+
+        /* Retrieve the created collections that need to be checked. */
+        collection_name = schema_table_name;
+        created_collections = parse_collections_schema(session, collection_name);
+
+        /* Allocate memory to the operations performed on the created collections. */
+        for (auto const &it : created_collections) {
+            std::map<int, std::string *> *map = new std::map<int, std::string *>();
+            collections[it] = map;
+        }
+
+        /*
+         * Build in memory the final state of each created collection according to the tracked
+         * operations.
+         */
+        collection_name = operation_table_name;
+        for (auto const &active_collection : created_collections)
+            parse_collections_operations(session, collection_name, active_collection, collections);
+
+        /* Check all tracked operations in memory against the database on disk. */
+        is_valid = check_reference(session, collections);
+
+        /* Check what has been saved on disk against what has been tracked. */
+        if (is_valid) {
+            for (auto const &collection : created_collections) {
+                is_valid = check_disk_state(session, collection, collections);
+                if (!is_valid) {
+                    debug_info("check_disk_state failed for collection " + collection, _trace_level,
+                      DEBUG_INFO);
+                    break;
+                }
+            }
+
+        } else
+            debug_info("check_reference failed!", _trace_level, DEBUG_INFO);
+
+        /* Clean the allocated memory. */
+        clean_memory(collections);
+
+        return is_valid;
+    }
+
+    /* Check what is present on disk against what has been tracked. */
+    bool
+    check_disk_state(WT_SESSION *session, const std::string &collection_name,
```

Comment:
Can we move this function down to the other function so they're not separated by parse functions.